domain-models-runtime/src/main/java/org/folio/rest/impl/TenantAPI.java

```diff
@@ -43,99 +45,16 @@
-              return;
-            }
-
-            String sqlFile = null;
-            try {
-              SchemaMaker sMaker = new SchemaMaker(tenantId, PostgresClient.getModuleName(),
-                  TenantOperation.DELETE, null, null);
-              sqlFile = sMaker.generateDDL();
-
-            } catch (Exception e1) {
-              handlers.handle(io.vertx.core.Future.failedFuture(e1.getMessage()));
-              log.error(e1.getMessage(), e1);
-              return;
-            }
-
-            log.info("Attempting to run delete script for: " + tenantId);
-            log.debug("GENERATED SCHEMA " + sqlFile);
-            /* connect as user in postgres-conf.json file (super user) - so that all commands will be available */
-            postgresClient(context).runSQLFile(sqlFile, true,
-                reply -> {
-                  try {
-                    // close clients because they still use the old oid of the dropped schema/role name,
-                    // they will fail when the same schema/role is recreated with a new oid.
-                    PostgresClient.closeAllClients(tenantId);
-                    String res = "";
-                    if(reply.succeeded()){
-                      res = new JsonArray(reply.result()).encodePrettily();
-                      if(reply.result().size() > 0){
-                        log.error("Unable to run the following commands during tenant delete: ");
-                        reply.result().forEach(log::error);
-                        handlers.handle(failedFuture(DeleteTenantResponse.respond400WithTextPlain(res)));
-                      }
-                      else {
-                        OutStream os = new OutStream();
-                        os.setData(res);
-                        handlers.handle(io.vertx.core.Future.succeededFuture(DeleteTenantResponse.respond204()));
-                      }
-                    }
-                    else {
-                      log.error(reply.cause().getMessage(), reply.cause());
-                      handlers.handle(failedFuture(DeleteTenantResponse
-                        .respond500WithTextPlain(reply.cause().getMessage())));
-                    }
-                  } catch (Exception e) {
-                    log.error(e.getMessage(), e);
-                    handlers.handle(failedFuture(DeleteTenantResponse
-                      .respond500WithTextPlain(e.getMessage())));
-                  }
-                });
-          });
-      } catch (Exception e) {
-        log.error(e.getMessage(), e);
-        handlers.handle(failedFuture(DeleteTenantResponse
-          .respond500WithTextPlain(e.getMessage())));
-      }
-    });
-  }
-
   Future<Boolean> tenantExists(Context context, String tenantId) {
     Promise<Boolean> promise = Promise.promise();
-    tenantExists(context, tenantId, promise.future());
+    tenantExists(context, tenantId, promise::handle);
     return promise.future();
```

Comment:
`::handle` is not needed:

    tenantExists(context, tenantId, promise);

This short form is also possible:

    return Future.future(promise -> tenantExists(context, tenantId, promise));

Reply:
ok. Thx